source/Halibut/Transport/SecureListener.cs

```diff
@@ -301,25 +306,40 @@ void SafelyRemoveClientFromTcpClientManager(TcpClient client, EndPoint clientNam
             }
         }
 
-        void SendFriendlyHtmlPage(Stream stream)
+        async Task SendFriendlyHtmlPage(Stream stream)
         {
             var message = getFriendlyHtmlPageContent();
             var headers = getFriendlyHtmlPageHeaders();
 
-            // This could fail if the client terminates the connection and we attempt to write to it
-            // Disposing the StreamWriter will close the stream - it owns the stream
-            using (var writer = new StreamWriter(stream, new UTF8Encoding(false)) { NewLine = "\r\n" })
+            if (asyncHalibutFeature.IsEnabled())
             {
-                writer.WriteLine("HTTP/1.0 200 OK");
-                writer.WriteLine("Content-Type: text/html; charset=utf-8");
-                writer.WriteLine("Content-Length: " + message.Length);
+                await stream.WriteControlLineAsync("HTTP/1.0 200 OK", cts.Token);
```

Comment:
While `WriteControlLineAsync` has the desired behaviour, it looks strange being used here. The reader may wonder what's going on as this is not control lines, just HTML. It may be worth adding a new extension for sending this over the stream.

Reply:
E.g. just `WriteLine`. It will duplicate code but I think that's ok!!

Reply:
Given that `WriteControlLineAsync` doesn't have anything "control line"-specific in the implementation, can we just rename the methods to `WriteLineAsync` and `WriteEmptyLineAsync`?

Duplication is fine if you want to keep the specific "Control Line" naming though.

Reply:
Leave `WriteControlLineAsync` as this can never change and just add a new extension that is WriteLineAsync etc.